### PR TITLE
Update uplink.types docstrings

### LINKED ISSUE #17: Add doc strings to argument annotation classes
Argument Annotation classes in `uplink/types.py` are missing class doc strings. To improve code documentation, we need to add doc strings to the following classes, adhering the [Google Style Guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) for consistency with the rest of the codebase:

- [ ] `uplink.types.Query`
- [ ] `uplink.types.QueryMap`
- [ ] `uplink.types.Header`
- [ ] `uplink.types.HeaderMap`
- [ ] `uplink.types.Field`
- [ ] `uplink.types.FieldMap`
- [ ] `uplink.types.Part`
- [ ] `uplink.types.PartMap`
- [ ] `uplink.types.Body`
- [ ] `uplink.types.Url`

### PR DESCRIPTION
**Pull request by FabioRosado**
Fixes #17

Changes proposed in this pull request:
- Adds docstrings to the classes:
   - uplink.types.Query
   - uplink.types.QueryMap
   - uplink.types.Header
   - uplink.types.HeaderMap
   - uplink.types.Field
   - uplink.types.FieldMap
   - uplink.types.Part
   - uplink.types.PartMap
   - uplink.types.Body
   - uplink.types.Url

I believe I've messed up a bit on the "Attribute/Args/Params" description of the classes. Some of the classes I did a bit of guessing of what they will be used for. With that in mind this is a first PR to try and get some feedback on my work.

Please let me know what needs to be improved (apologies if it's a lot of things).

Attention: @prkumar

### CODE REVIEW

File: AUTHORS.rst [hunk 1]
```diff
@@ -6,3 +6,4 @@ Contributors
 ************
 - Kareem Moussa (`@itstehkman <https://github.com/itstehkman>`_)
 - Brandon Milton (`@brandonio21 <https://github.com/brandonio21>`_)
+- Fabio Rosado (`@fabiorosado <https://github.com/fabiorosado>`_)
```
Comment (prkumar) [inline AUTHORS.rst]: 👍

File: uplink/types.py [hunk 2]
```diff
@@ -289,6 +302,16 @@ def modify_request(self, request_builder, value):
...
 
 class QueryMap(TypedArgument):
+    """Sets a mapping of query arguments.
+
+    If the API you are using accepts multiple query arguments,
+    you can include them all in your function method by using the
+    format: `<query argument>:uplink.QueryMap`
+
+    Args:
+        **params: A number of query argument accepted by the API.
```
Comment (prkumar) [inline uplink/types.py]: Like you did with `HeaderMap`, I think "mapping" would be more descriptive than "number" here.

File: uplink/types.py [hunk 3]
```diff
@@ -366,48 +436,91 @@ def modify_request(self, request_builder, value):
...
         request_builder.info["data"] = value
 
 
 class Url(ArgumentAnnotation):
+    """Sets a dynamic URL.
+
+    Provides the URL at runtime as a method argument. Drop the
+    decorator parameter path from `uplink.get` and annotate the
+    corresponding argument with `uplink.Url`
+
```
Comment (prkumar) [inline uplink/types.py]: Looks good!

Nit: Remove extra whitespace

File: uplink/types.py [hunk 4]
```diff
@@ -269,6 +269,19 @@ def modify_request(self, request_builder, value):
 
 
 class Query(NamedArgument):
+    """Sets one dynamic query parameter.
+
+    Queries the API using it's defined argument. You can include it in
+    your function methody by using the format: <query argument>:uplink.Query.
```
Comment (prkumar) [inline uplink/types.py]: This paragraph looks good overall. Here, I would replace the first sentence with "This annotation turns argument values into URL query parameters". Also, there's a typo on `methody` in the second sentence (the word `methody` has a nice ring to it, though 😄).

File: uplink/types.py [hunk 5]
```diff
@@ -301,29 +324,62 @@ def modify_request(cls, request_builder, value):
 
 
 class Header(NamedArgument):
+    """Pass a header as a method argument at runtime.
+
+    While decorator.headers are meant to be used as a decorator,
+    this argument (uplink.Header) is meant to be passed as a method
+    argument on a function created by you.
+
+    Used as: uplink.Header("<header parameter>")
```
Comment (prkumar) [inline uplink/types.py]: Change "header parameter" to "header key"

**Comment by FabioRosado** [comment]
I've updated the docstrings following your suggestions @prkumar thank you so much for the review!
I also agree that methody does have a nice ring to it! 😄

**Comment by prkumar** [comment]
I think the coverage drop is due to an issue with creating the coverage reports on Python 3.3. #31 fixed that, but I think your `documentation` branch was cut before #31 was merged into `master`, so I don't think we need to worry about it.

**Pull request merged**